gdal/frmts/raw/byndataset.cpp

```diff
@@ -193,13 +194,40 @@ int BYNDataset::Identify( GDALOpenInfo *poOpenInfo )
         hHeader.nVDatum    < -1 || hHeader.nVDatum    > 3 ||
         hHeader.nDatum     < -1 || hHeader.nDatum     > 1 ||
         hHeader.nDescrip   < -1 || hHeader.nDescrip   > 3 ||
-        hHeader.nByteOrder < -1 || hHeader.nByteOrder > 1 ||
-        std::abs( static_cast<GIntBig>(hHeader.nSouth) - ( hHeader.nDLat / 2 ) ) > BYN_MAX_LAT ||
-        std::abs( static_cast<GIntBig>(hHeader.nNorth) - ( hHeader.nDLat / 2 ) ) > BYN_MAX_LAT ||
-        std::abs( static_cast<GIntBig>(hHeader.nWest)  - ( hHeader.nDLon / 2 ) ) > BYN_MAX_LON ||
-        std::abs( static_cast<GIntBig>(hHeader.nEast)  - ( hHeader.nDLon / 2 ) ) > BYN_MAX_LON )
+        hHeader.nByteOrder < -1 || hHeader.nByteOrder > 1 )
         return FALSE;
 
+    /* Move from cell center to upper left, lower right */
+
+    try
+    {
+        hHeader.nSouth = std::abs( hHeader.nSouth - ( hHeader.nDLat / 2 ) );
+        hHeader.nNorth = std::abs( hHeader.nNorth + ( hHeader.nDLat / 2 ) );
+        hHeader.nWest  = std::abs( hHeader.nWest  - ( hHeader.nDLon / 2 ) );
+        hHeader.nEast  = std::abs( hHeader.nEast  + ( hHeader.nDLon / 2 ) );
+    }
+    catch( const CPLSafeIntOverflow& )
```

Comment:
the exception will never be emitted unless you use the CPLSM() macro to wrap the values. I'd suggest just keeping the cast to GIntBig I did. This is simpler

Reply:
OK, but in that case, there will be an extra cast for each operation to assign the value back to GInt32.

Reply:
never mid that my last comment.